main.cpp

```diff
@@ -92,27 +109,79 @@ void link_executable(const Buildfile& buildfile, const vector<string>& cpp_files
     }
 }
 
-int main() {
+bool is_created(const string &filename){
+
+    // tests for the existence of the file with success being a 0
+    if(access(filename.c_str(), F_OK) == 0)
+    {
+        return true;
+    }
+    return false;
+}
+
+bool parse_commandline_args(int argc, char** &argv)
+{
+    struct option longOpts[] = {
+            { "help", no_argument, nullptr, 'h' },
+            { nullptr, 0, nullptr, '\0' }
+        };
+
+    int option, optionIndex;
+
+    while((option = getopt_long(argc, argv, "h", longOpts, &optionIndex))!= -1)
+    {
+        switch(option)
+        {
+            case 'h':
+            {
+                cout<<help_header;
+                return false;
+            }
+
+            default:
+            {
+                return false;
```

Comment:
Personally, I am a fan of using c++ exceptions to indicate error conditions. This allows the calling convention to not include error statuses and usually simplifies control logic. In this case I don't think it will have a big impact beyond encouraging future consistency. Long story short, are you opposed to throwing a std::runtime_exception here with the message equal to help_msg? This would allow the caller to do a simple ```if(!parse_command_line(blah)) return; // false return value means do not continue with execution``` and a universal handling of error cases.

Also, it avoids the subtle but potentially important bug where the return code of the program is 0 even if an invalid option was specified.